model/map/src/main/java/org/keycloak/models/map/authSession/MapRootAuthenticationSessionAdapter.java

```diff
@@ -83,6 +93,18 @@ public AuthenticationSessionModel getAuthenticationSession(ClientModel client, S
     public AuthenticationSessionModel createAuthenticationSession(ClientModel client) {
         Objects.requireNonNull(client, "The provided client can't be null!");
 
+        Set<MapAuthenticationSessionEntity> authenticationSessions = entity.getAuthenticationSessions();
+        if (authenticationSessions != null && authenticationSessions.size() >= authSessionsLimit) {
+            String tabId = authenticationSessions.stream().min(TIMESTAMP_COMPARATOR).map(MapAuthenticationSessionEntity::getTabId).orElse(null);
```

Comment:
Should we optimize this? Not sure if now, or create a follow-up task. But this way with each authentication session we need to load and sort all authentication sessions from the store. Maybe we can introduce a new method in the `RootAuthenticationSessionEntity` something like `removeOldest()`, in JPA this can be solved by one SQL command.

Reply:
Not sure how it is in JPA store, but in Hot Rod we load the whole entity anyway, don't we? I guess, we would also need a method for counting number of auth sessions. I'm leaning towards doing optimization later, when we encounter a bottleneck in current approach.

Reply:
Ok then, let's leave it as is for now and revisit later. Also, we will see how the discussion in the team chat I started will end up. If we won't do the optimizations in JPA we can leave this as is now.